Adds the snow machine

Citinited opened the pull request:
**What this PR does:**

This PR adds the snow machine, a new kind of buildable machine that cools the surrounding area, then spawns snow clouds if it's cold enough, then makes snow. Snow can be cleared away using a shovel.

Click on the snow to make a snowball. Snowballs deal 10 stamina damage per hit (roughly 30% of a disabler beam)

It can snow on unsimulated turfs if anyone wants to set one up in Centcomm.

Admins can set the `infinite_snow` var to non-zero to make it snow without needing any reagents

**Pics**

![snowmachine](https://user-images.githubusercontent.com/18459142/50038127-6e36b080-0012-11e9-94ed-321ab0bcff63.gif)

![snow](https://user-images.githubusercontent.com/18459142/50053194-7da81d80-0128-11e9-8fc6-a1d5d5277099.gif)

:cl:
add: Adds the snow machine. Fill it up with water, turn it on, and watch it create a winter wonderland all on its own! For best results set your room's air alarm to the lowest value it can go to.
add: The snow machine is orderable from cargo during Christmas time.
tweak: You can use a shovel to clear away generated snow. Scrooge.
/:cl:

TDSSS commented:
Does it cool stuff low enough to be dangerous for certain crew (slimes)?

Citinited commented:
It cools down to -10C, which is just warm enough that humans don't take damage. Unsure of slimes though, will have to test

TDSSS commented on code/game/machinery/constructable_frame.dm:
```diff
@@ -317,6 +317,16 @@ to destroy them and players will be able to make replacements.
+	frame_desc = "Requires 1 Matter Bin and 1 Micro Laser."
+	req_components = list(
+							/obj/item/stock_parts/matter_bin = 1,
+							/obj/item/stock_parts/micro_laser = 1)
```
Laser to make snow? Stupid confusing NT hightech 😕

Citinited replied:
The laser is there for more efficient power use. Do you think a capacitor would be better here?

TDSSS commented on code/game/objects/effects/snowcloud.dm:
```diff
@@ -0,0 +1,138 @@
+
+/obj/effect/snow/proc/melt()
+	var/turf/simulated/T = get_turf(src)
+	T.MakeSlippery() //Yes it'll runtime if it spawns on an unsimulated turf. But that shouldn't happen as the only way to spawn this is to have a snowcloud, and that can only spawn on simulated turfs
```
I wouldn't rely on that to avoid runtimes. What if someone later wants to reuse the snow effect, or an admin wants to add snow clouds to CC when the escape shuttle arrives ,and so on? Personally, I'd add a simple check in here.

Citinited replied:
Hmm, I'll add a check here then.

Citinited replied:
On second thoughts I removed snow making the ground slippery when it melts because I can see that being supremely annoying (especially as you can't see the wet floor overlay if more snow piled on top of it)

TDSSS commented on code/game/objects/effects/snowcloud.dm:
```diff
@@ -0,0 +1,138 @@
+	name = "snowball"
+	desc = "Get ready for a snowball fight!"
+	force = 0
+	throwforce = 2
```
Might want to just have snowballs do stamina damage, so you can have snowball fights without people needing healing afterwards?

Citinited replied:
Good idea, will change

Citinited commented:
More changes committed. 
- Added it as a festive-only crate from cargo
- It now takes any reagents and can make snow regardless of what it's filled with. (Because of the magic of Christmas 🎄

@TDSSS I've changed the things you suggested, and can comfirm it does hurt slimes if they don't take any protections. I'd suggest some kind of cold-weather gear or coffee to counteract the cooling 😄

Quantum-M commented:
Is there any way to clean the snow? Or does it "melt" after the temperature in the tile reaches a certain high point?

Citinited commented:
It will have a chance to melt once the temperature rises above freezing. Maybe I should add a snow shovel item, though (for all those scrooges out there)

Citinited commented:
Update: my todo list is done, and I have created a fancy-ass sprite for this.

![snowmachine](https://user-images.githubusercontent.com/18459142/50037436-0e87d780-0009-11e9-9623-aaf3ea8e77fc.gif)

![snow](https://user-images.githubusercontent.com/18459142/50037438-16477c00-0009-11e9-8554-d163283b9ee5.gif)

KasparoVy commented:
Holy fuck, let it snow let it snow let it snow! That's lovely

Citinited commented:
This is no longer WIP; review & merge-ready 🎉 

Changes:
- You can clear away snow with a shovel now
- It will snow on unsimulated turfs now (but won't change their temperature, obviously)
- Sprite has slightly less big wheels, and an open panel sprite.

Mitchs98 commented:
Kinda lame slimes won't be able to do Christmas snow things if they want to survive without drinking copious amounts of coffee :v. Otherwise, looks neat.

Citinited commented:
Mm. I could've made it like how it affects unsimulated turfs (i.e. just spawn snow clouds), but that wouldn't have been as cool as how it's implemented now :smile:

In any case, if people keep raising the coldness as an issue I can always PR a modification
